lib/puppet/util/windows/sid.rb

```diff
@@ -97,6 +97,29 @@ def octet_string_to_sid_object(bytes)
     end
     module_function :octet_string_to_sid_object
 
+    def ads_to_sid_object(ads_object)
```

Comment:
nit: given this returns Puppet::Util::Windows::SID::Principal objects, should the method name be `ads_to_principal_object` ?

Reply:
Yeah, that could make more sense... Was being consistent with naming of existing methods

Reply:
So we have `name_to_sid_object` and `octet_string_to_sid_object` that return `Principal` objects already. If we're going to rename this method, then I think we should rename the others as well, and maybe add aliases back to the original names for backwards compat.

I'd do that in a separate refactor commit after this one...

Reply:
Added refactor commit for this

Reply:
Added 9da5ed7 for this, but it introduces a bit of churn at 102 lines...